getpaid/processor.py

```diff
@@ -0,0 +1,81 @@
+from abc import abstractmethod, ABC
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+
+class BaseProcessor(ABC):
+    display_name = None
+    accepted_currencies = None
+    logo_url = None
+    slug = None  # for friendly urls
+    method = 'GET'
+    template_name = None
+
+    def __init__(self, payment):
+        self.payment = payment
+        self.path = payment.backend
+        if self.slug is None:
+            self.slug = payment.backend  # no more Mr. Friendly :P
+
+    def get_form(self, post_data):
+        """
+        Only used if the payment processor requires POST requests.
+        Generates a form only containg hidden input fields.
+        """
+        from getpaid.forms import PaymentHiddenInputsPostForm
```

Comment:
This look really off, why there is import inside method? Is there any good justification for that?

If this is circular import shouldn't this be just:
```
from getpaid import forms
[...]
    def get_form(self, post_data):
        return forms.PaymentHiddenInputsPostForm(items=post_data)
```

Reply:
Nope, won't import.